### ISSUE TITLE
Refreshed token doesn't contain capabilities for all channels

### ISSUE THREAD

**Issue by sacOO7** [ably/laravel-broadcaster#55]
- Customer has reported an issue where `refreshed token` doesn't contain all channels, causing existing channels to reconnect again. 
> The current issue is that when Ably hits the token expiry time, it makes multiple token refresh calls. Although the current token has all the necessary capabilities, the server sometimes responds with a token containing only the channel_name provided in the payload. This causes the token to be marked as invalid, which then triggers another refresh cycle to obtain a token with proper capabilities.
- Link to internal conversation => https://ably-real-time.slack.com/archives/CURL4U2FP/p1733135454672739

**Comment by sacOO7** [comment]
A temporary solution for now would be to set ABLY_TOKEN_EXPIRY in `.env` file.
```
ABLY_TOKEN_EXPIRY=21600
```

**Comment by sacOO7** [comment]
- Generally, ably requests the client to renew the token 30 seconds before expiry as per [RTN22](https://sdk.ably.com/builds/ably/specification/main/features/#RTN22)
- Link to internal conversation -> https://ably-real-time.slack.com/archives/C8SPU4589/p1724236742837859

**Comment by graphem** [comment]
Yeah seems I am experiecing this and this is killing my server, as all the clients seems to make tons of request again.

**Comment by graphem** [comment]
@sacOO7 Any news on merging this and doing a release?

**Comment by sacOO7** [comment]
Currently, this is kept on hold and will be picked up soon, can you try increasing `TOKEN_EXPIRY` for the time being to say 6 hrs, wdyt

**Comment by graphem** [comment]
Ok thanks! Seems the token expiry is not making a difference so maybe my issues is something different.

**Comment by sacOO7** [comment]
@graphem okay, make sure to report whatever issue you are facing 👍 
Thanks !

**Comment by sacOO7** [comment]
@graphem can you track pr log userId or ip address of users requesting tokens?
Also, if you dont mind, how many users use your app at a time? and how many private/presence they use on average?

**Comment by graphem** [comment]
I am not using the presence channels. I am using very few channels basically the app might have up to 30 different private channel and might have 2-3 public channels. And we are running like 5 apps per server. I am going to try to log the entries to get more details but I was simply observing the access logs and it was just insane traffic. We have around 2000 users concurrent per servers right now. We use decent baremetal servers which can take the traffic but this is not sustainable right now as we grow.

**Comment by sacOO7** [comment]
Okay, it will be great if you can analyse logs and check which requests users are sending concurrently? 
I am confident you must have battle tested app on the local. It will also be great to analyse if requests are coming from mobile  browsers that are causing any such issues? I think you can also use available AI tools to analyse logs.

**Comment by graphem** [comment]
Yes it is a tough one cause we are not able to reproduce locally our in our staging environment. I am going to analyse the log further today and see if I can find more info

**Comment by sacOO7** [comment]
@graphem thanks. We will be eagerly waiting for your analysis 👍

**Comment by graphem** [comment]
Is there a way to send private files as I don't want to share sensitive info here

**Comment by sacOO7** [comment]
Sure, you can open ticket at https://ably.com/support and share information there 👍

**Comment by graphem** [comment]
Ok cool I already have a ticket, I will gather more info

**Comment by graphem** [comment]
Actually seems the token expiry seems to make a difference today, so it might be that issue

**Comment by sacOO7** [comment]
Great 👍 
you can also try increasing the timeout value to say 12 or 24 hrs. Let us know about your findings.

**Comment by graphem** [comment]
@sacOO7 What is the setting for the timeout?

**Comment by sacOO7** [comment]
I didn't get your question? You mean token expiry or something else

**Comment by graphem** [comment]
> Great 👍 you can also try increasing the timeout value to say 12 or 24 hrs. Let us know about your findings.

You mentioned a timeout value here

**Comment by sacOO7** [comment]
Ohh I meant this -> https://github.com/ably/laravel-broadcaster/issues/55#issuecomment-2521091010

**Comment by sacOO7** [comment]
Btw, were you able to achieve desired behaviour?

**Comment by graphem** [comment]
Oh yeah sorry I increased it, our server load is 200% better this morning, this seems to have done the trick

**Comment by graphem** [comment]
Ok thanks! Yeah it is much better looking forward for the merge here. Thanks!

**Comment by sacOO7** [comment]
Need to add extra checks as per discussion => https://github.com/ably/specification/pull/259#discussion_r1890228871
